src/Compilers/Core/MSBuildTask/IAnalyzerConfigFilesHostObject.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    [ComImport]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    [Guid("31891ED8-BEB5-43BF-A90D-9E7E1CE9BA84")]
+    public interface IAnalyzerConfigFilesHostObject
+    {
+        bool SetAnalyzerConfigFiles(ITaskItem[] analyzerConfigFiles);
+        bool SetPotentialAnalyzerConfigFiles(ITaskItem[] potentialAnalyzerConfigfiles);
+    }
```

Comment:
Why do we need this vs. having the IDE just look at the compiler command line?

Reply:
If a .editorconfig file is added or removed the project system will need to re-run a design-time build to figure out the new set of applicable files. As such the project system needs to know which files to watch.

Reply:
Sure but can't you also get that from the command line arguments we're already passing to the IDE?

Reply:
1. csproj/msvbprj do not currently have the ability to parse the command line arguments. There's no reason to add that ability when I can just get the paths straight from MSBuild.

2. The "potential" analyzer config files represent all the places the build task looked for .editorconfig files but didn't find them. There's no compiler command line argument that corresponds to that concept, so I have to pass them in some other way.

We will eventually get rid of the host object and then these silly interfaces go away. However, I don't know if that's going to happen for 16.0 or 16.1 so I can't depend on it.